tools/Raven.Debug/LogTrafficWatch/TrafficWatchReplay.cs

```diff
@@ -0,0 +1,313 @@
+                try
+                {
+                    return new UriBuilder(original) {
+                        Host = _host,
+                        Port = _port,
+                        Scheme = _schema
+                    }.Uri;
+
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+
+            private static string ConvertJsonElement(JsonElement e)
+            {
+                return e.ValueKind switch
+                {
+                    JsonValueKind.Number => e.ToString(),
+                    JsonValueKind.Array => e.ToString(),
+                    JsonValueKind.Object => e.ToString(),
+                    _ => $"\"{e}\""
+                };
+            }
+        }
+
+        public static int RegexIndexOf(string str, string pattern)
+        {
+            var m = Regex.Match(str, pattern);
+            return m.Success ? m.Index : -1;
+        }
+
+
+        private static void InitializeHttpClient(string certPath, string certPass)
+        {
+            if (string.IsNullOrEmpty(certPath) == false)
+            {
+                HttpClient = new HttpClient(new HttpClientHandler
+                {
+                    ClientCertificates =
+                    {
+                        new X509Certificate2(certPath, certPass, X509KeyStorageFlags.MachineKeySet)
+                    }
+                });
+                return;
+            }
+
+            HttpClient = new HttpClient();
+        }
+
+        private static void ExitAndPrintError(Exception ex, string errorText)
+        {
+            Console.ForegroundColor = ConsoleColor.Red;
+            Console.WriteLine(errorText);
+            Console.ResetColor();
+            if (ex != null)
+            {
+                Console.WriteLine();
+                Console.WriteLine(ex);
+            }
+
+            Environment.Exit(2);
+        }
+
+        public class TrafficWatchChange
+        {
+            public DateTime TimeStamp { get; set; }
+            public int RequestId { get; set; }
+            public string HttpMethod { get; set; }
+            public long ElapsedMilliseconds { get; set; }
+            public int ResponseStatusCode { get; set; }
+            public string RequestUri { get; set; }
+            public string AbsoluteUri { get; set; }
+            public string DatabaseName { get; set; }
+            public string CustomInfo { get; set; }
+            public TrafficWatchChangeType Type { get; set; }
+            public string ClientIP { get; set; }
+            public string CertificateThumbprint { get; set; }
+            public string TrafficWatchType { get; set; }
+        }
+
+        public void Stop()
+        {
+            Console.WriteLine("Stop collection traffic watch. Exiting...");
+            Dispose();
+            Environment.Exit(0);
+        }
+
+        public void Dispose()
```

Comment:
do we need that?

Reply:
Done by adding [assembly: InternalsVisibleTo("Raven.Debug, PublicKey=xxx")] to ..src\Raven.Client\Properties\AssemblyInfo.Windows.cs and ..src\Raven.Client\Properties\AssemblyInfo.Linux.cs

Reply:
Restore it for HttpClient disposing of.